src/utils.php

```diff
@@ -32,3 +32,15 @@ function pathToUri(string $filepath): string {
     $filepath = implode('/', array_map('urlencode', explode('/', $filepath)));
     return 'file:///' . $filepath;
 }
+
+/**
+ * Transforms URI into file path
+ * 
+ * @param string $uri
+ * @return string
+ */
+function uriToPath(string $uri)
+{
+    $filepath = urldecode(parse_url($uri)['path']);
+    return strpos($filepath, ':') === false ? $filepath : str_replace('/', '\\', $filepath);
```

Comment:
This is a little brittle. There are also implementations that use `|` instead of the colon on windows. You don't need to test against that, but it would be better to simply do `str_replace('/', DIRECTORY_SEPERATOR, $filepath)`

https://en.wikipedia.org/wiki/File_URI_scheme#Windows